app/src/main/java/org/mozilla/fenix/home/collections/CollectionMenu.kt

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home.collections
+
+import android.content.Context
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.DropdownMenu
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.selector.normalTabs
+import mozilla.components.feature.tab.collections.TabCollection
+import org.mozilla.fenix.R.string
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.home.sessioncontrol.CollectionInteractor
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Menu shown for a [TabCollection].
+ *
+ * @see [DropdownMenu]
+ *
+ * @param showMenu Whether this is currently open and visible to the user.
+ * @param menuItems List of options shown.
+ * @param onDismissRequest Called when the user chooses a menu option or requests to dismiss the menu.
+ */
+@Composable
+fun CollectionMenu(
+    showMenu: Boolean,
+    menuItems: List<CollectionMenuItemData>,
+    onDismissRequest: () -> Unit
+) {
+    DisposableEffect(LocalConfiguration.current.orientation) {
+        onDispose { onDismissRequest() }
+    }
+
+    // DropdownMenu uses the medium shape from MaterialTheme.
+    // Override it's corner radius to be the same 8.dp as in mozac_browser_menu_corner_radius
+    MaterialTheme(shapes = MaterialTheme.shapes.copy(medium = RoundedCornerShape(8.dp))) {
+        DropdownMenu(
+            expanded = showMenu,
+            onDismissRequest = { onDismissRequest() },
+            modifier = Modifier
+                .background(color = FirefoxTheme.colors.layer2)
+        ) {
+            for (item in menuItems) {
+                DropdownMenuItem(
+                    onClick = {
+                        onDismissRequest()
+                        item.onClick()
+                    },
+                ) {
+                    Text(
```

Comment:
does this `Text` need any text size attributes to be set here?

Reply:
Don't think so. This menu follows the other used for homescreen composables like the one [for RecentVisits](https://github.com/mozilla-mobile/fenix/blob/f924f744348006e2cb01fb172486fc7bfb1b67dc/app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentlyVisited.kt#L328-L335), [for RecentBookmarks](https://github.com/mozilla-mobile/fenix/blob/f924f744348006e2cb01fb172486fc7bfb1b67dc/app/src/main/java/org/mozilla/fenix/home/recentbookmarks/view/RecentBookmarks.kt#L205-L212) or [for RecentTabs](https://github.com/mozilla-mobile/fenix/blob/f924f744348006e2cb01fb172486fc7bfb1b67dc/app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabs.kt#L488-L495).

Reply:
gotcha gotcha. I just get a little anxious seeing key parameters not initialized 😅

Reply:
I know 😅. A reusable `Menu` / `MenuItem` might help with this, with them hardcoding a certain style to also ensure nothing changes in how the menu looks if some defaults change in Compose.